### PR TITLE
Fix code blocks not rendering on documentation site

### PR DESCRIPTION
**Pull request by camerondubas**
Code blocks on https://ember-concurrency.com/docs/tutorial are not rendering. This is due to deprecated code in the `CodeSnippet` component's template (provided by [ember-code-snippet](https://github.com/ef4/ember-code-snippet)).

To solve this, I bumped to thes latest `ember-code-snippet`, which required adding and configuring [ember-prism](https://github.com/shipshapecode/ember-prism) as well as creating a local `CodeSnippet` component as this is no longer provided by `ember-code-snippet` directly.

Additionally, the calculation of the max height for the template-toggle component broke so I refactored to use ResizeObserver.

Here are a couple of updated screenshots.  Note that the styling is a bit different from before. This can be customized, however, I felt this could be addressed in a follow-up PR.
image
image

Closes https://github.com/machty/ember-concurrency/issues/521

### CODE REVIEW

File: tests/dummy/app/components/code-snippet/component.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+export default class CodeTemplateToggleComponent extends Component {
+  tagName = '';
+
+  @computed('name')
+  get snippet() {
+    return getCodeSnippet(this.name);
+  }
+
+  get language() {
+    if (this.snippet.language === 'handlebars') {
```
Comment (camerondubas) [inline tests/dummy/app/components/code-snippet/component.js:14]: ember-prism expects hbs files to use the language "markup".  This feels a bit hacky. Would love a better solution

File: tests/dummy/app/components/code-template-toggle/component.js [hunk 2]
```diff
@@ -10,40 +9,45 @@ export default class CodeTemplateToggleComponent extends Component {
...
-        })
-      );
-
-      let toggle = element.querySelector('.code-template-toggle');
+    const element = document.getElementById(this.id);
+    if (!element) {
+      return;
+    }
 
+    let resizeObserver = this._createResizeObserver((maxHeight) => {
```
Comment (maxfierke) [inline tests/dummy/app/components/code-template-toggle/component.js:21]: do we need to call `disconnect()` on the observer in `willDestroyElement`?

**Review by maxfierke — COMMENTED** [review]
Looks great! One question about observer cleanup.

(Looks like the embroider jobs are failing since #515 was merged, so ignore for now.)

File: tests/dummy/app/components/code-template-toggle/component.js (hunk 2 above)
Comment (camerondubas) [reply tests/dummy/app/components/code-template-toggle/component.js]: Good point! Updated it with the cleanup in `willDestroyElement`

**Pull request merged**